sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -103,12 +125,40 @@ protected String visitIsNotNull(String v) {
     return v + " IS NOT NULL";
   }
 
+  protected String visitStartsWith(String l, String r) {
+    String value = r.replaceAll("'", "");
+    return l + " LIKE '" + value + "%'";
+  }
+
+  protected String visitEndsWith(String l, String r) {
+    String value = r.replaceAll("'", "");
```

Comment:
what if the string value really contains `'`? I think we should be more explicit here: `r.substring(1, r.length - 1)`